src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -899,37 +914,114 @@ await CallWithRequestTracing(async () =>
                 }
                 else
                 {
-                    ConfigurationSnapshot snapshot;
+                    // Load snapshot data
+                    SnapshotReference snapshotReference = new SnapshotReference(loadOption.SnapshotName, client, cancellationToken);
+                    Dictionary<string, ConfigurationSetting> resolvedSettings = await Resolve(snapshotReference).ConfigureAwait(false);
 
-                    try
-                    {
-                        snapshot = await client.GetSnapshotAsync(loadOption.SnapshotName).ConfigureAwait(false);
-                    }
-                    catch (RequestFailedException rfe) when (rfe.Status == (int)HttpStatusCode.NotFound)
+                    foreach (KeyValuePair<string, ConfigurationSetting> resolvedSetting in resolvedSettings)
                     {
-                        throw new InvalidOperationException($"Could not find snapshot with name '{loadOption.SnapshotName}'.", rfe);
+                        data[resolvedSetting.Key] = resolvedSetting.Value;
                     }
+                }
+            }
 
-                    if (snapshot.SnapshotComposition != SnapshotComposition.Key)
+            return data;
+        }
+        private string ParseSnapshotReference(ConfigurationSetting setting)
+        {
+            if (setting == null || string.IsNullOrWhiteSpace(setting.Value))
+            {
+                return null;
+            }
+
+            try
+            {
+                Utf8JsonReader reader = new Utf8JsonReader(System.Text.Encoding.UTF8.GetBytes(setting.Value));
```

Comment:
```suggestion
                var reader = new Utf8JsonReader(System.Text.Encoding.UTF8.GetBytes(setting.Value));
```

Please also follow this convention in other place: var xxx = new SomeClass();